integration_test/pg/storage_test.exs

```diff
@@ -139,6 +139,19 @@ defmodule Ecto.Integration.StorageTest do
     def change, do: :ok
   end
 
+  defmodule CreateTestSchemaMigration do
+    use Ecto.Migration
+
+    def change do
+      execute TestRepo.create_prefix("test_schema")
+
+      create table(:schema_migrations, prefix: "test_schema") do
+        add :version, :integer
+        add :inserted_at, :timestamp
+      end
+    end
+  end
```

Comment:
```suggestion
```

I'm pretty sure we can get rid of this migration and use the `run_psql` function at the beginning of the test

```elixir
run_psql(~s[create schema "test_schema"])
```

Reply:
I tried removing the migration and adding `run_psql(~s[create schema "test_schema"])` but unfortunately, that resulted in the following error so I've kept the migration to add the schema for now. 
```elixir
20:20:56.994 [error] Could not create schema migrations table. This error usually happens due to the following:

       * The database does not exist
       * The "schema_migrations" table, which Ecto uses for managing
         migrations, was defined by another library
       * There is a deadlock while migrating (such as using concurrent
         indexes with a migration_lock)

     To fix the first issue, run "mix ecto.create" for the desired MIX_ENV.

     To address the second, you can run "mix ecto.drop" followed by
     "mix ecto.create", both for the desired MIX_ENV. Alternatively you may
     configure Ecto to use another table and/or repository for managing
     migrations:

         config :ecto_sql, Ecto.Integration.PoolRepo,
           migration_source: "some_other_table_for_schema_migrations",
           migration_repo: AnotherRepoForSchemaMigrations

     The full error report is shown below.
```

Reply:
The `run_psql` command is connecting to the `storage_mgt` database by default but the repo running the migrations is connecting to `ecto_test`. We can fix it this way:

```elixir
test "structure dump and load with migrations table on multiple database schemas" do
    # Create the test_schema schema
    run_psql(~s[CREATE SCHEMA test_schema], [PoolRepo.config()[:database]])

    # Run migrations
    version = @base_migration + System.unique_integer([:positive])
    :ok = Ecto.Migrator.up(PoolRepo, version, Migration, log: false)
    :ok = Ecto.Migrator.up(PoolRepo, version, Migration, log: false, prefix: "test_schema")

    config = Keyword.put(TestRepo.config(), :migration_prefixes, ["public", "test_schema"])
    {:ok, path} = Postgres.structure_dump(tmp_path(), config)
    contents = File.read!(path)

    assert contents =~ ~s[INSERT INTO public."schema_migrations" (version) VALUES (#{version})]
    assert contents =~ ~s[INSERT INTO test_schema."schema_migrations" (version) VALUES (#{version})]
  after
    run_psql(~s[DROP SCHEMA test_schema], [PoolRepo.config()[:database]])
  end
```

Reply:
Ah got it! Thanks for the explanation 😄